### PR TITLE
ISSUE #697 Decomissioned Role, User and Camera types

### LINKED ISSUE #697: Mongo Driver Upgrade - decommissioning
### Description

Upgrade the mongo driver; [product ticket](https://github.com/3drepo/3D-Repo-Product-Team/issues/561) with further details.

This issue is for the decommissioning part of that ticket, the purpose of which is to clean up the bouncer codebase so we only focus on refactoring the bits we still use.

### Goals

- [ ] Remove legacy functions of bouncer that are no longer part of its use case

### Tasks

- [x] Ensure 462 is merged
- [x] Remove diff functionality
- [x] Remove role and user functionality
- [x] Remove cameras
- [x] Remove project settings
- [x] Remove gridfs
- [x] Remove matrix logic from generateFederation 
- [x] Remove getFile from client
- [x] Remove GLTF exporter
- [x] Remove any remaining (publicly) unused db operations (countItemsInCollection, removeCollection, removeDocument, hasDatabase, getNameOfAdminDatabase, getDatabasesWithProjects, getDatabases, fetchDatabases)

### PR DESCRIPTION
**Pull request by sebjf**
This fixes #697

#### Description

This PR performs a clean-up of the bouncer codebase, removing unused types and methods reflective of its role as a monolithic application, so that its public API (e.g. RepoController) is better aligned with its current and near-future use case (i.e. as a pipeline component operating on existing documents managed by the .io backend).

The motivation is to reduce the amount of code we have to refactor for Mongo - especially as, not being used, much of that logic won't be exercised or tested properly.

The PR has removed the following types and functions

`GridFSFileHandler`
`GLTFModelExport`
`CameraNode`
`RepoUser`
`RepoNode`
`removeDocument` [1]
`removeCollections`
`countItemsInCollection`
`dropCollections`
`getDatabasesWithProjects`
`getDatabases`
`fetchDatabases`
`getNameOfAdminDatabase`
`getStandardDatabaseRoles`
`upsertDocument` [1]
`fetchCollections`
`getProjects`

[1] In some cases, e.g. these, the method has been removed from the public API but remains for the database handlers, as they are used internally. Other methods have been removed right through.

Finally, the `getFile` commands been removed from 3drepobouncerClient, and the admin tool has been removed from tools.

This PR only has one actual "code change", which is that the `projectExists` function for the unit tests now uses functions from `repo_test_database_info.h`, instead of using a `RepoController` instance.

Otherwise, changes are limited to the signiatures

### CODE REVIEW

**Comment by sebjf** [comment]
Hi @carmenfan, while on our deleting expedition should we remove GLTFModelExport as well? (Its non-zero cost to maintain because it interacts with the web buffers code, and is very out of date - I was just looking and realised it also relies on .io hosting shaders that are no longer there!)

**Comment by carmenfan** [comment]
@sebjf I'd say so yes 🙈

**Review by carmenfan — CHANGES_REQUESTED** [review]
I think we can get rid of the assimp exporter as well

the rest of the comments I leave it up to you 😆 I don't feel like backwards compatibility is too much of a concern since with unity bundles removed we are unlikely to be reading existing data with our current use case for bouncer?

File: bouncer/src/repo/manipulator/modelconvertor/export/repo_model_export_assimp.cpp [hunk 1]
```diff
@@ -191,30 +191,6 @@ aiNode* AssimpModelExport::constructAiSceneRecursively(
 						meshIndices.push_back(index);
 					}
 					break;
-
```
[the remark this answers is not in the document]
Comment (sebjf) [reply bouncer/src/repo/manipulator/modelconvertor/export/repo_model_export_assimp.cpp:194]: Done!

File: bouncer/src/repo/core/model/bson/repo_bson_ref.h [hunk 2]
```diff
@@ -42,7 +42,9 @@ namespace repo {
 				const static std::string REPO_REF_TYPE_UNKNOWN;
 
 				enum class RefType {
-					GRIDFS, FS, UNKNOWN
+					GRIDFS, // GRIDFS is no longer supported but may exist in the database
```
[the remark this answers is not in the document]
Comment (sebjf) [reply bouncer/src/repo/core/model/bson/repo_bson_ref.h]: Done!

File: bouncer/src/repo/core/model/repo_model_global.h [hunk 3]
```diff
@@ -187,7 +187,7 @@
 
 #define REPO_NODE_TYPE_ANIMATION		"animation"
 #define REPO_NODE_TYPE_BONE				"bone"
-#define REPO_NODE_TYPE_CAMERA			"camera"
+#define REPO_NODE_TYPE_CAMERA			"camera" // No longer supported in Bouncer but may exist in previous imports
```
[the remark this answers is not in the document]
Comment (sebjf) [reply bouncer/src/repo/core/model/repo_model_global.h]: Done!

**Review by carmenfan — APPROVED** [review]

**Pull request merged**